### PR TITLE
The Great Rename 

### PR DESCRIPTION
**Pull request by ghost**
Sponge (coremod) becomes SpongeForge.
Let us celebrate by breaking all references and links.

Please check if I missed occurences that should be fixed.

### CODE REVIEW

**Comment by Spongy** [comment]
Setting up PR reference, please wait...

File: source/server/getting-started/migrating.rst [hunk 1]
```diff
@@ -15,10 +15,10 @@ Overview
 The purpose of the articles within this section is to help current server owners to migrate from other server platforms
 to Sponge.
 
-Migrating to Sponge (coremod)
-=============================
+Migrating to SpongeForge
+========================
 
-To migrate to a Forge server running the Sponge coremod, select the platform you are migrating from below.
+To migrate to a Forge server running the SpongeForge, select the platform you are migrating from below.
```
Comment (Inscrutable) [inline source/server/getting-started/migrating.rst]: "server running SpongeForge" (remove the *the*).

File: source/contributing/api/index.rst [hunk 2]
```diff
@@ -6,7 +6,7 @@ Overview
 ========
 
 The Sponge API is a community-developed API allowing developers to write plugins for Minecraft servers that run an
-implementation of the Sponge API, such as the Sponge coremod.
+implementation of the Sponge API, such as the SpongeForge or SpongeVanilla.
```
Comment (Inscrutable) [inline source/contributing/api/index.rst]: "such as SpongeForge or SpongeVanilla." Please remove unwanted *the*.

**Comment by Inscrutable** [comment]
You may also want to incorporate the draft I had for #343 too, as this PR tinkers with that page heavily.

**Pull request merged**